src/index/TextBlockWriter.cpp

```diff
@@ -0,0 +1,118 @@
+#include "TextBlockWriter.h"
+
+#include "index/TextIndexReadWrite.h"
+
+template <typename WordTextVecView, typename EntityTextVecView>
+void TextBlockWriter<WordTextVecView, EntityTextVecView>::
+    calculateAndWriteTextBlocks(size_t nofWordPostingsInTextBlock) {
+  AD_CONTRACT_CHECK(
+      nofWordPostingsInTextBlock > 0,
+      "Number of word postings in text block has to be larger than zero.");
+  for (auto&& chunk :
+       ::ranges::views::chunk(wordTextVecView_, nofWordPostingsInTextBlock)) {
+    for (const auto& row : chunk) {
+      currentWordTextVecWordIndex_ = row[0].getWordVocabIndex();
+      addWordPosting(row);
+    }
+    addBlock();
+  }
+}
+
+// _____________________________________________________________________________
+template <typename WordTextVecView, typename EntityTextVecView>
+void TextBlockWriter<WordTextVecView, EntityTextVecView>::addBlock() {
+  // Add all co-occurring entities to entityPostings_
+  addEntityPostingsUpToWordIndex();
+
+  // This is possible since the wordPostings are filled in ascending WordIndex
+  // order.
+  WordVocabIndex minWordIndexOfBlock = std::get<1>(wordPostings_.front());
+  WordVocabIndex maxWordIndexOfBlock = std::get<1>(wordPostings_.back());
+
+  // Sort both posting vectors by TextRecordIndex, wordIndex (or entityIndex),
+  // score
+  ql::ranges::sort(wordPostings_);
+  ql::ranges::sort(entityPostings_);
+
+  // Filter out duplicate entity postings. The reason for duplicates is the
+  // following case: Different words that co-occur with the same entities in
+  // the same text record appear in the same text block.
+  // An example would be a text record "He <Newton> helped" and the words
+  // "he" and "helped" are put in the same block.
+  entityPostings_.erase(std::ranges::unique(entityPostings_).begin(),
+                        entityPostings_.end());
```

Comment:
don't use `std::ranges` . Unfortunately `ql::ranges::unique` doesn't work, because std::ranges and ::ranges (range-v3) have different return types for `unique`...
so just `::ranges::unique` could work, or `std::unique` with begin and end.